java/plasma/src/test/java/org/apache/arrow/plasma/PlasmaClientTest.java

```diff
@@ -0,0 +1,161 @@
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class PlasmaClientTest {
+
+    private String storeSuffix = "/tmp/store";
+
+    private Process storeProcess;
+
+    private int storePort;
+
+    private ObjectStoreLink pLink;
+
+
+    public PlasmaClientTest() throws Exception{
+        try {
+            String plasmaStorePath = System.getenv("PLASMA_STORE");
+            if(plasmaStorePath == null) {
+                throw new Exception("Please set plasma store path in env PLASMA_STORE");
+            }
+
+            this.startObjectStore(plasmaStorePath);
+            System.loadLibrary("plasma_java");
+            pLink = new PlasmaClient(this.getStoreAddress(), "", 0);
+        }
+        catch (Throwable t) {
+            cleanup();
+            throw t;
+        }
+
+    }
+
+    private Process startProcess(String[] cmd) {
+        ProcessBuilder builder;
+        List<String> newCmd = Arrays.stream(cmd).filter(s -> s.length() > 0).collect(Collectors.toList());
+        builder = new ProcessBuilder(newCmd);
+        Process p = null;
+        try {
+            p = builder.start();
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
+        }
+        System.out.println("Start process " + p.hashCode() + " OK, cmd = " + Arrays.toString(cmd).replace(',', ' '));
+        return p;
+    }
+
+    private void startObjectStore(String plasmaStorePath) {
+        int occupiedMemoryMB = 10;
+        long memoryBytes = occupiedMemoryMB * 1000000;
+        int numRetries = 10;
+        Process p = null;
+        while (numRetries-- > 0) {
+            int currentPort = java.util.concurrent.ThreadLocalRandom.current().nextInt(0, 100000);
+            String name = storeSuffix + currentPort;
+            String cmd = plasmaStorePath + " -s " + name + " -m " + memoryBytes;
+
+            p = startProcess(cmd.split(" "));
+
+            if (p != null && p.isAlive()) {
+                try {
+                    TimeUnit.MILLISECONDS.sleep(100);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+                if (p.isAlive()) {
+                    storePort = currentPort;
+                    break;
+                }
+            }
+        }
+
+
+        if (p == null || !p.isAlive()) {
+            throw new RuntimeException("Start object store failed ...");
+        } else {
+            storeProcess = p;
+            System.out.println("Start object store success");
+        }
+    }
+
+    private void cleanup() {
+        if (storeProcess != null && killProcess(storeProcess)) {
+            System.out.println("Kill plasma store process forcely");
+        }
+    }
+
+    private static boolean killProcess(Process p) {
+        if (p.isAlive()) {
+            p.destroyForcibly();
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public void doTest() {
```

Comment:
can we add tests for "contains" and "hash" here?

Reply:
If I remember correctly, 'contains' will use plasma_manager which is not exist here.

Reply:
Both contains and hash are just functions of the plasma_store, I think you meant the "Info" call

Reply:
Yes, you are right, i just test the contains and hash , both work well. I will add the test case soon